### PR TITLE
luminous:  mds,osd,mon,msg: use intrusive_ptr for holding Connection::priv

### PR DESCRIPTION
**Pull request by athanatos**
Fixes: https://tracker.ceph.com/issues/24360

### CODE REVIEW

**Comment by smithfarm** [comment]
jenkins test signed

**Comment by yuriw** [comment]
[wip-yuri5-testing-2019-09-30-1944-luminous](https://trello.com/c/oisaBLUN/700-wip-yuri5-testing-2019-09-30-1944-luminous)

File: src/mds/MDSDaemon.cc [hunk 1]
```diff
@@ -1364,15 +1366,15 @@ bool MDSDaemon::ms_verify_authorizer(Connection *con, int peer_type,
...
+      con->set_priv(RefCountedPtr{s, false});
       s->connection = con;
       if (mds_rank) {
         mds_rank->kick_waiters_for_any_client_connection();
       }
     } else {
       dout(10) << " existing session " << s << " for " << s->info.inst << " existing con " << s->connection
 	       << ", new/authorizing con " << con << dendl;
-      con->set_priv(s->get());
+      con->set_priv(RefCountedPtr{s, false});
```
Comment (batrick) [inline src/mds/MDSDaemon.cc]: This causes the fs failures. This backport also needs #22384.

@vshankar

Comment (vshankar) [reply src/mds/MDSDaemon.cc]: thanks patrick -- I'll post the backport.

Comment (smithfarm) [reply src/mds/MDSDaemon.cc]: follow-on luminous backport PR: https://github.com/ceph/ceph/pull/31049

**Pull request merged**